Automattic/woocommerce-payments
# Add compatibility between Multi-Currency and WooCommerce Name Your Price

Fixes #3507

Note: Description and notes below are from [original PR](https://github.com/Automattic/woocommerce-payments/pull/3578). I created a PR in our repo for easier testing, and I also added tests after @helgatheviking and I worked out all the bugs.

#### Changes proposed in this Pull Request

Introduce a class that:
filters the raw minimum, maximum, and suggested prices to other currencies.
and
convert prices for NYP products from entered currency to current currency


#### Testing instructions

* Create a simple product with a minimum price
* On frontend, switch to a different currency. See min price change.
* Add the product to the cart.
* Go to cart, change currency. Cart price should now be converted from the price in the entered currency to the current currency.

-------------------


**Post merge**



- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) - Does not apply.


This ended up a little more complex than I started out with... My thinking what that we should avoid converting from a currency to the default store currency, to then convert it back to the initial currency. Something like: USD > EUR > USD. While I didn't run into any errors myself, I've run into rounding issues with things like that in the past. 

There's also a point [here](https://github.com/Automattic/woocommerce-payments/pull/3578/files#diff-6671eecca02fb2be4da9826c4286eae7a3f89eea2fb14c2963c041ebd1f4be19R106) where I tell NYP to use the price setters to manipulate the price of the cart item product object. I was hoping I could just set the `$cart_item['nyp']` value, but that's not possible without a change to the NYP plugin. I might make the change anyway, but then there's a change here that would be required (remove above mentioned line and run that callback at an earlier priority) _and_ it would only work with a new version of NYP. This approach will work with any version.
